### ISSUE TITLE
Add clone/stamp tool

### ISSUE THREAD

**Issue by dsizzle** [HaikuArchives/ArtPaint#519]
Since we deleted a tool and need a new one, ArtPaint could use a clone/stamp tool

- Select a reference/sampling point with right click
- Drawing with left button copies the area from the reference point 
- Should have "Use current brush" option
- Should have "All layers" vs "Current layer" for sampling
- Maybe a "jitter" option to jiggle the sample area while drawing?

**Comment by humdingerb** [comment]
Plus the popular "Pressure" slider.

> Drawing with left button copies the area from the reference point

One thing I like from Photoshop's clone tool is that the reference point moves along with the mouse pointer. Can we have that, too?

**Comment by dsizzle** [comment]
> One thing I like from Photoshop's clone tool is that the reference point moves along with the mouse pointer. Can we have that, too?

Yeah, I think that's necessary otherwise the tool isn't super useful. Doesn't photoshop have 2 modes?  
Something like:
1. reset to reference point at the start of every stroke
2. maintain current relative offset for new strokes

**Comment by humdingerb** [comment]
I had a quick look at the extremely old Photoshop lite or whatever I have installed at work.
It has two modes, one to stamp a fixed pattern, and one cloning another part of the image. I've only used the second, clone mode.

You hold ALT, the pointer changes to a crosshair, you move the mouse to the spot that will become the clone source, you left-click and let go of ALT. Next time you left-click the source pixels are inserted, maintaining the relative offset while you move. While cloning, the source spot is marked with a sort of ghosty shadow-brush that is following your brush movement.
SHIFT limits movement horizontally/vertically while cloning.

Something like that would be nice...
